mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/core/MigrateApiConfigFlows.java

```diff
@@ -0,0 +1,198 @@
+
+  RamlModelBuilder builder = null;
+  RamlModelResult result = null;
+
+  @Override
+  public String getDescription() {
+    return "Migrate ApiConfig flow definitions";
+  }
+
+  public MigrateApiConfigFlows() {
+    this.setAppliedTo(XPATH_SELECTOR);
+  }
+
+  @Override
+  public void execute(Element element, MigrationReport report) throws RuntimeException {
+
+    // resolve config-ref
+    String apiConfig = null;
+
+    final Element node = getApplicationModel().getNode(XPATH_SELECTOR_ROUTER);
+    if (node != null && node.getAttribute("config-ref") != null) {
+      apiConfig = node.getAttribute("config-ref").getValue();
+
+    }
+    if (apiConfig != null) {
+      // API Kit Router found
+      addNameSpace(CORE_EE_NAMESPACE, EE_NAMESPACE_SCHEMA, element.getDocument());
+
+
+      final String flowName = element.getAttributeValue("name");
+
+      if (flowName.endsWith(":" + apiConfig)) {
+
+        List<String> uriParams = getResourceByFlowName(flowName.replaceAll(":" + apiConfig, ""), report, element);
+        if (uriParams != null && uriParams.size() > 0) {
+          Element transformElement = new Element("transform", CORE_EE_NAMESPACE);
+          transformElement.removeContent();
+          final Element variablesElement = new Element("variables", CORE_EE_NAMESPACE);
+          variablesElement.removeContent();
+          transformElement.addContent(variablesElement);
+
+          for (String name : uriParams) {
+            final Element variableElement = new Element("set-variable", CORE_EE_NAMESPACE);
+            variableElement.setAttribute("variableName", name);
+            variableElement.setContent(new CDATA("attributes.uriParams." + name));
+            variablesElement.addContent(variableElement);
+          }
+
+          // Add at index 0 to be the first element
+          element.addContent(0, transformElement);
+        }
+      }
+    }
+  }
+
+  private List<String> getResourceByFlowName(final String flowName, MigrationReport migrationReport, Element element)
+      throws RuntimeException {
+    List<String> uriParams = null;
+    File ramlFile = null;
+    if (builder == null) {
+      final String ramlLocation = getApplicationModel().getPomModel().get().getArtifactId() + ".raml";
+      ramlFile =
+          getApplicationModel().getProjectBasePath().resolve(MULE_4_API_FOLDER + File.separator + ramlLocation).toFile();
+      builder = new RamlModelBuilder();
+      result = builder.buildApi(ramlFile);
+    }
+
+    if (result.getApiV10() == null) {
+
+      migrationReport.report("raml.invalid", element, element,
+                             result.getValidationResults().toString());
+      throw new RuntimeException("Ramlfile invalid: " + result.getValidationResults().toString());
+    }
+
+
+    final List<Resource> resources = result.getApiV10().resources();
+    boolean found = false;
+    for (Resource resource : resources) {
```

Comment:
I need to review this logic further, I'm wondering if we really need to use the API spec or we can simply trust that the APIkit flow name correctly identifies the URI params. So, just analyzing the flow name and adding as params everything in between `{}` as those should be the params.